app/models/user.rb

```diff
@@ -0,0 +1,5 @@
+class User < ApplicationRecord
+  validates :username, presence: true, length: { maximum: 4 }
+  validates :email, presence: true
+  validates :password, presence: true
```

Comment:
These two could be merged as `validates :email, :password, presence: true`

Reply:
Thanks for the suggestion. It's been fixed